modules/host/dependencies.go

```diff
@@ -0,0 +1,79 @@
+package host
+
+import (
+	"errors"
+	"net"
+	"os"
+
+	"github.com/NebulousLabs/Sia/persist"
+)
+
+// Fake errors that get returned when a simulated failure of a dependency is
+// desired for testing.
+var (
+	mockErrListen       = errors.New("simulated Listen failure")
+	mockErrLoadFile     = errors.New("simulated LoadFile failure")
+	mockErrMkdirAll     = errors.New("simulated MkdirAll failure")
+	mockErrNewLogger    = errors.New("simulated NewLogger failure")
+	mockErrOpenDatabase = errors.New("simulated OpenDatabase failure")
+)
+
+// These interfaces define the Host's dependencies. Mocking implementation
+// complexity can be reduced by defining each dependency as the minimium
+// possible subset of the real dependency.
+type (
+	// dependencies defines all of the dependencies of the Host.
+	dependencies interface {
+		// Listen gives the host the ability to receive incoming connections.
+		Listen(string, string) (net.Listener, error)
```

Comment:
so, this is sort of a special case because none of the dependencies here are implemented by existing objects. For example, instead of using the logger's methods (e.g. `Println`, `Close`), you use `NewLogger`. And the other modules aren't included in this interface at all. If you wanted to combine their methods into this single interface, you might have some collisions.